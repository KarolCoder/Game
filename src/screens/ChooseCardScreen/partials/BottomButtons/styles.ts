import {Button} from 'react-native-paper';
import styled from 'styled-components/native';

export const BottomButton = styled(Button)`
  padding: 4px 8px;
  flex: 1;
`;
