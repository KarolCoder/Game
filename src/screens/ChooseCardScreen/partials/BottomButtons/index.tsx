import {Spacer} from '@/components';
import React from 'react';
import {Row} from '../../styles';
import {BottomButton} from './styles';

interface BottomButtonsProps {
  isDisabled?: boolean;
  onLeftPress: () => void;
  onRightPress: () => void;
}

export const BottomButtons: React.FC<BottomButtonsProps> = ({
  isDisabled,
  onLeftPress,
  onRightPress,
}) => {
  return (
    <Row>
      <BottomButton
        mode="contained"
        disabled={isDisabled}
        onPress={onLeftPress}>
        Reset cards
      </BottomButton>
      <Spacer width={24} />
      <BottomButton
        mode="contained"
        disabled={isDisabled}
        onPress={onRightPress}>
        Go to start
      </BottomButton>
    </Row>
  );
};
