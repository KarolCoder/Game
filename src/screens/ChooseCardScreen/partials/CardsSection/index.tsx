import {Spacer} from '@/components';
import React from 'react';
import {Title, useTheme} from 'react-native-paper';
import {Row} from '../../styles';
import {Card, SelectTitle} from './styles';

interface CardsSectionProps {
  setSelectedCard: React.Dispatch<React.SetStateAction<number | undefined>>;
  refreshData: () => void;
  getNextPage: () => Promise<void>;
  firstValueDisplayed?: string;
  secondValueDisplayed?: string;
  resourcesFromData?: {
    firstResource: string | string[] | undefined;
    secondResource: string | string[] | undefined;
  };
  nextPage?: string;
  status?: string;
}

export const CardsSection: React.FC<CardsSectionProps> = ({
  resourcesFromData,
  setSelectedCard,
  refreshData,
  getNextPage,
  firstValueDisplayed,
  secondValueDisplayed,
  nextPage,
  status,
}) => {
  const {colors} = useTheme();
  return (
    <>
      <SelectTitle textColor={colors.primary}>Choose card</SelectTitle>
      <Row>
        <Card
          disabled={
            (!!resourcesFromData?.firstResource &&
              !!resourcesFromData?.secondResource) ||
            status === 'fetching'
          }
          onPress={() => {
            setSelectedCard(0);
            if (!nextPage) {
              refreshData();
            } else {
              getNextPage();
            }
          }}>
          <Title>{firstValueDisplayed ? firstValueDisplayed : '?'}</Title>
        </Card>
        <Spacer width={24} />
        <Card
          disabled={
            (!!resourcesFromData?.firstResource &&
              !!resourcesFromData?.secondResource) ||
            status === 'fetching'
          }
          onPress={() => {
            setSelectedCard(1);
            if (!nextPage) {
              refreshData();
            } else {
              getNextPage();
            }
          }}>
          <Title>{secondValueDisplayed ? secondValueDisplayed : '?'}</Title>
        </Card>
      </Row>
    </>
  );
};
