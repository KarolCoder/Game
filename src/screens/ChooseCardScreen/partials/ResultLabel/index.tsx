import {BigText} from '../../styles';
import React from 'react';
import {useTheme} from 'react-native-paper';
import {GameResult} from '@/utils/types';

interface ResultLabelProps {
  result: GameResult;
}

export const ResultLabel: React.FC<ResultLabelProps> = ({result}) => {
  const {colors} = useTheme();

  return result === 'Winner' ? (
    <BigText textColor={colors.primary}>Congratualtions, you won!!!</BigText>
  ) : result === 'Loser' ? (
    <BigText textColor={colors.primary}>You lose, loser :D</BigText>
  ) : result === 'Tie' ? (
    <BigText textColor={colors.primary}>Tie, try again</BigText>
  ) : !result ? (
    <BigText textColor={colors.primary}>
      Couldn't resolve winner please go to start
    </BigText>
  ) : null;
};
