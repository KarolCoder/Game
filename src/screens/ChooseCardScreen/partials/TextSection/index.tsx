import React from 'react';
import {useTheme} from 'react-native-paper';
import {BigText, Row, SmallText} from '../../styles';
import {CategoryBox} from './styles';

interface TextSectionProps {
  leftTopText?: string;
  rightTopText?: string;
  bottomLeftText?: string | number;
  bottomRightText?: string | number;
}

export const TextSection: React.FC<TextSectionProps> = ({
  leftTopText = '',
  rightTopText = '',
  bottomLeftText = '',
  bottomRightText = '',
}) => {
  const {colors} = useTheme();
  return (
    <Row>
      <CategoryBox>
        <SmallText textColor={colors.primary}>{leftTopText}</SmallText>
        <BigText textColor={colors.primary}>{bottomLeftText}</BigText>
      </CategoryBox>
      <CategoryBox>
        <SmallText textColor={colors.primary}>{rightTopText}</SmallText>
        <BigText textColor={colors.primary}>{bottomRightText}</BigText>
      </CategoryBox>
    </Row>
  );
};
