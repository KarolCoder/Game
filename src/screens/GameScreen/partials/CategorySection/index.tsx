import {Spacer} from '@/components';
import {GameCategory} from '@/utils/types';
import React from 'react';
import {useTheme} from 'react-native-paper';
import {Row, SelectTitle} from '../../styles';
import {OptionButton} from './styles';

interface CategorySectionProps {
  activeCategory?: GameCategory;
  setActiveCategory: (category?: GameCategory | undefined) => void;
}

export const CategorySection: React.FC<CategorySectionProps> = ({
  setActiveCategory,
  activeCategory,
}) => {
  const {colors} = useTheme();
  return (
    <>
      <SelectTitle textColor={colors.primary}>Choose category</SelectTitle>
      <Row>
        <OptionButton
          testID="option button 1"
          isActive={activeCategory === 'People'}
          mode="contained"
          onPress={() => setActiveCategory('People')}>
          People
        </OptionButton>
        <Spacer width={24} />
        <OptionButton
          testID="option button 2"
          isActive={activeCategory === 'Starships'}
          mode="contained"
          onPress={() => setActiveCategory('Starships')}>
          Starships
        </OptionButton>
      </Row>
    </>
  );
};
