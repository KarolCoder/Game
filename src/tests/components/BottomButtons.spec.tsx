import {BottomButtons} from '@/screens/ChooseCardScreen/partials/BottomButtons';
import {fireEvent, render} from '@testing-library/react-native';
import React from 'react';

import renderer from 'react-test-renderer';

describe('bottom buttons', () => {
  it('renders correctly', async () => {
    const fn = jest.fn();
    const component = <BottomButtons onLeftPress={fn} onRightPress={fn} />;
    const result = renderer.create(component).toJSON();
    const {getByTestId} = render(component);

    expect(result).toMatchSnapshot();
    getByTestId('bottom first button');
    getByTestId('bottom second button');
  });

  it('disabled', async () => {
    const fn = jest.fn();
    const component = (
      <BottomButtons onLeftPress={fn} onRightPress={fn} isDisabled />
    );
    const result = renderer.create(component).toJSON();
    const {getByTestId} = render(component);

    expect(result).toMatchSnapshot();
    const firstButton = getByTestId('bottom first button');
    const secondButton = getByTestId('bottom second button');
    fireEvent.press(firstButton);
    expect(fn).toBeCalledTimes(0);
    fireEvent.press(secondButton);
    expect(fn).toBeCalledTimes(0);
  });

  it('pressed', async () => {
    const fn = jest.fn();
    const component = <BottomButtons onLeftPress={fn} onRightPress={fn} />;
    const result = renderer.create(component).toJSON();
    const {getByTestId} = render(component);
    expect(result).toMatchSnapshot();
    const firstButton = getByTestId('bottom first button');
    const secondButton = getByTestId('bottom second button');
    fireEvent.press(firstButton);
    expect(fn).toBeCalledTimes(1);
    fireEvent.press(secondButton);
    expect(fn).toBeCalledTimes(2);
  });
});
