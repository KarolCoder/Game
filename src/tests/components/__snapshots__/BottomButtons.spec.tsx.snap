// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bottom buttons disabled 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
        "justifyContent": "space-evenly",
        "paddingBottom": 24,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 24,
        "width": "100%",
      },
    ]
  }
>
  <View
    collapsable={false}
    style={
      Object {
        "backgroundColor": "rgba(0, 0, 0, 0.12)",
        "borderColor": "transparent",
        "borderRadius": 4,
        "borderStyle": "solid",
        "borderWidth": 0,
        "elevation": 0,
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "minWidth": 64,
        "paddingBottom": 4,
        "paddingLeft": 8,
        "paddingRight": 8,
        "paddingTop": 4,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "disabled": true,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 4,
          },
        ]
      }
      testID="bottom first button"
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            },
            undefined,
          ]
        }
      >
        <Text
          numberOfLines={1}
          selectable={false}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              Object {
                "textAlign": "left",
              },
              Array [
                Object {
                  "letterSpacing": 1,
                  "marginHorizontal": 16,
                  "marginVertical": 9,
                  "textAlign": "center",
                },
                undefined,
                Object {
                  "textTransform": "uppercase",
                },
                Object {
                  "color": "rgba(0, 0, 0, 0.32)",
                  "fontFamily": "System",
                  "fontWeight": "500",
                },
                Object {
                  "fontFamily": "System",
                  "fontWeight": "500",
                },
                undefined,
              ],
            ]
          }
        >
          Reset cards
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "height": 0,
          "width": 24,
        },
      ]
    }
    width={24}
  />
  <View
    collapsable={false}
    style={
      Object {
        "backgroundColor": "rgba(0, 0, 0, 0.12)",
        "borderColor": "transparent",
        "borderRadius": 4,
        "borderStyle": "solid",
        "borderWidth": 0,
        "elevation": 0,
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "minWidth": 64,
        "paddingBottom": 4,
        "paddingLeft": 8,
        "paddingRight": 8,
        "paddingTop": 4,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "disabled": true,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 4,
          },
        ]
      }
      testID="bottom second button"
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            },
            undefined,
          ]
        }
      >
        <Text
          numberOfLines={1}
          selectable={false}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              Object {
                "textAlign": "left",
              },
              Array [
                Object {
                  "letterSpacing": 1,
                  "marginHorizontal": 16,
                  "marginVertical": 9,
                  "textAlign": "center",
                },
                undefined,
                Object {
                  "textTransform": "uppercase",
                },
                Object {
                  "color": "rgba(0, 0, 0, 0.32)",
                  "fontFamily": "System",
                  "fontWeight": "500",
                },
                Object {
                  "fontFamily": "System",
                  "fontWeight": "500",
                },
                undefined,
              ],
            ]
          }
        >
          Go to start
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`bottom buttons pressed 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
        "justifyContent": "space-evenly",
        "paddingBottom": 24,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 24,
        "width": "100%",
      },
    ]
  }
>
  <View
    collapsable={false}
    style={
      Object {
        "backgroundColor": "#6200ee",
        "borderColor": "transparent",
        "borderRadius": 4,
        "borderStyle": "solid",
        "borderWidth": 0,
        "elevation": 2,
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "minWidth": 64,
        "paddingBottom": 4,
        "paddingLeft": 8,
        "paddingRight": 8,
        "paddingTop": 4,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 0.75,
          "width": 0,
        },
        "shadowOpacity": 0.24,
        "shadowRadius": 1.5,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 4,
          },
        ]
      }
      testID="bottom first button"
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            },
            undefined,
          ]
        }
      >
        <Text
          numberOfLines={1}
          selectable={false}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              Object {
                "textAlign": "left",
              },
              Array [
                Object {
                  "letterSpacing": 1,
                  "marginHorizontal": 16,
                  "marginVertical": 9,
                  "textAlign": "center",
                },
                undefined,
                Object {
                  "textTransform": "uppercase",
                },
                Object {
                  "color": "#ffffff",
                  "fontFamily": "System",
                  "fontWeight": "500",
                },
                Object {
                  "fontFamily": "System",
                  "fontWeight": "500",
                },
                undefined,
              ],
            ]
          }
        >
          Reset cards
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "height": 0,
          "width": 24,
        },
      ]
    }
    width={24}
  />
  <View
    collapsable={false}
    style={
      Object {
        "backgroundColor": "#6200ee",
        "borderColor": "transparent",
        "borderRadius": 4,
        "borderStyle": "solid",
        "borderWidth": 0,
        "elevation": 2,
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "minWidth": 64,
        "paddingBottom": 4,
        "paddingLeft": 8,
        "paddingRight": 8,
        "paddingTop": 4,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 0.75,
          "width": 0,
        },
        "shadowOpacity": 0.24,
        "shadowRadius": 1.5,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 4,
          },
        ]
      }
      testID="bottom second button"
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            },
            undefined,
          ]
        }
      >
        <Text
          numberOfLines={1}
          selectable={false}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              Object {
                "textAlign": "left",
              },
              Array [
                Object {
                  "letterSpacing": 1,
                  "marginHorizontal": 16,
                  "marginVertical": 9,
                  "textAlign": "center",
                },
                undefined,
                Object {
                  "textTransform": "uppercase",
                },
                Object {
                  "color": "#ffffff",
                  "fontFamily": "System",
                  "fontWeight": "500",
                },
                Object {
                  "fontFamily": "System",
                  "fontWeight": "500",
                },
                undefined,
              ],
            ]
          }
        >
          Go to start
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`bottom buttons renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
        "justifyContent": "space-evenly",
        "paddingBottom": 24,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 24,
        "width": "100%",
      },
    ]
  }
>
  <View
    collapsable={false}
    style={
      Object {
        "backgroundColor": "#6200ee",
        "borderColor": "transparent",
        "borderRadius": 4,
        "borderStyle": "solid",
        "borderWidth": 0,
        "elevation": 2,
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "minWidth": 64,
        "paddingBottom": 4,
        "paddingLeft": 8,
        "paddingRight": 8,
        "paddingTop": 4,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 0.75,
          "width": 0,
        },
        "shadowOpacity": 0.24,
        "shadowRadius": 1.5,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 4,
          },
        ]
      }
      testID="bottom first button"
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            },
            undefined,
          ]
        }
      >
        <Text
          numberOfLines={1}
          selectable={false}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              Object {
                "textAlign": "left",
              },
              Array [
                Object {
                  "letterSpacing": 1,
                  "marginHorizontal": 16,
                  "marginVertical": 9,
                  "textAlign": "center",
                },
                undefined,
                Object {
                  "textTransform": "uppercase",
                },
                Object {
                  "color": "#ffffff",
                  "fontFamily": "System",
                  "fontWeight": "500",
                },
                Object {
                  "fontFamily": "System",
                  "fontWeight": "500",
                },
                undefined,
              ],
            ]
          }
        >
          Reset cards
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "height": 0,
          "width": 24,
        },
      ]
    }
    width={24}
  />
  <View
    collapsable={false}
    style={
      Object {
        "backgroundColor": "#6200ee",
        "borderColor": "transparent",
        "borderRadius": 4,
        "borderStyle": "solid",
        "borderWidth": 0,
        "elevation": 2,
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "minWidth": 64,
        "paddingBottom": 4,
        "paddingLeft": 8,
        "paddingRight": 8,
        "paddingTop": 4,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 0.75,
          "width": 0,
        },
        "shadowOpacity": 0.24,
        "shadowRadius": 1.5,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 4,
          },
        ]
      }
      testID="bottom second button"
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            },
            undefined,
          ]
        }
      >
        <Text
          numberOfLines={1}
          selectable={false}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              Object {
                "textAlign": "left",
              },
              Array [
                Object {
                  "letterSpacing": 1,
                  "marginHorizontal": 16,
                  "marginVertical": 9,
                  "textAlign": "center",
                },
                undefined,
                Object {
                  "textTransform": "uppercase",
                },
                Object {
                  "color": "#ffffff",
                  "fontFamily": "System",
                  "fontWeight": "500",
                },
                Object {
                  "fontFamily": "System",
                  "fontWeight": "500",
                },
                undefined,
              ],
            ]
          }
        >
          Go to start
        </Text>
      </View>
    </View>
  </View>
</View>
`;
